lib/memory_profiler/results.rb

```diff
@@ -133,23 +136,38 @@ def pretty_print(io = $stdout, **options)
           METRICS.each do |metric|
             NAMES.each do |name|
               scale_data = metric == "memory" && options[:scale_bytes]
-              dump "#{type} #{metric} by #{name}", self.send("#{type}_#{metric}_by_#{name}"), io, scale_data
+              dump "#{type} #{metric} by #{name}", self.send("#{type}_#{metric}_by_#{name}"), io, scale_data, normalize_paths
             end
           end
         end
       end
 
       io.puts
-      dump_strings(io, "Allocated", strings_allocated, limit: options[:allocated_strings])
+      dump_strings(io, "Allocated", strings_allocated, normalize_paths, limit: options[:allocated_strings])
       io.puts
-      dump_strings(io, "Retained", strings_retained, limit: options[:retained_strings])
+      dump_strings(io, "Retained", strings_retained, normalize_paths, limit: options[:retained_strings])
 
       io.close if io.is_a? File
     end
 
+    def normalize_path(path)
+      @normalize_path ||= {}
+      @normalize_path[path] ||= begin
+        if %r!(/gems/.*)*/gems/(?<gemname>[^/]+)(?<rest>.*)! =~ path
+          "#{gemname}#{rest}"
+        elsif %r!ruby/2\.[^/]+/(?<stdlib>[^/.]+)(?<rest>.*)! =~ path
+          "ruby/lib/#{stdlib}#{rest}"
+        elsif %r!(?<app>[^/]+/(bin|app|lib))(?<rest>.*)! =~ path
+          "#{app}#{rest}"
+        else
+          path
+        end
+      end
+    end
+
     private
 
-    def dump_strings(io, title, strings, limit: nil)
+    def dump_strings(io, title, strings, normalize_paths, limit: nil)
```

Comment:
I wonder if this should be `normalize_paths: false, limit: nil`  which makes caller more explicit. 

I guess we are already a bit of a mish mash internally so it would be a bit of a big cleanup to take care of making all of this consistent

Reply:
Since this is a `private` method, it shouldn't matter much at all.
However, when cleaning up, I feel it'd be best to resort to an `opts = {}` so that future additions do not increase the line-length..